### PR TITLE
Added writeFile function fix #54

### LINKED ISSUE #54: make match data write to file

### PR DESCRIPTION
**Pull request by atian7064**
Writefile function is now added

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/SecondFragment2.java [hunk 1]
```diff
@@ -71,12 +76,36 @@ public View onCreateView(
...
     }
 
+    /*Suppressed due to server having no way to get the dynamic path
+        TODO: Add matchNumber to file name once added
+    */
+    @SuppressLint("SdCardPath")
+    public void writeFile() throws IOException {
+        new File("/sdcard/Documents/ScoutingData/").mkdirs();
+
+        File dataFile = new File("/sdcard/Documents/ScoutingData/data.txt");
```
Comment (ConfettiRaindrop) [inline app/src/main/java/com/example/myapplication/SecondFragment2.java:86]: This looks like the file will always be called data.txt - is there a way to differentiate between the matches for them? Or does the computer have to look inside the file to find the difference?

Comment (atian7064) [reply app/src/main/java/com/example/myapplication/SecondFragment2.java:86]: I forgot to add the match number 💀

File: app/src/main/java/com/example/myapplication/SecondFragment2.java [hunk 2]
```diff
@@ -71,12 +76,36 @@ public View onCreateView(
...
+        dataFile.createNewFile();
+        PrintWriter pw = new PrintWriter(dataFile);
+        pw.print(UserModel.getMatchData().returnAllData());
+        pw.close();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Files.deleteIfExists(Paths.get("/sdcard/Documents/ScoutingData/lock.txt"));
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/SecondFragment2.java:92]: Do we need a lock file? Right now there is no locking being done in this project.

Comment (atian7064) [reply app/src/main/java/com/example/myapplication/SecondFragment2.java:75]: Yes, the computer via adb will access the files and transfer them. The absence of lock.txt means that the computer needs to update the changes. The computer will create lock.txt when it is done.

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 3]
```diff
@@ -48,16 +48,30 @@ public enum Harmony {
...
 
-
+    public String returnAllData() {
+        //TODO: Consider use of JSON library
+        return "{\"teamNumber\":\"" + teamNumber + "\"," + "\"matchNumber\":\"" + matchNumber + "\"," + "\"workingAuto\":\"" + workingAuto + "\"," + "\"moveOutOfZone\":\""
+                + moveOutOfZone + "\"," + "\"ampAuto\":\"" + ampAuto + "\"," + "\"speakerAuto\":\"" + speakerAuto + "\","
+                + "\"isBroke\":\"" + broke + "\"," + "\"isDefense\":\"" + defense + "\"," + "\"ampTeleop\":\"" + ampTeleop + "\"," + "\"amplifiedSpeaker\":\"" + amplifiedSpeaker + "\","
+                + "\"speakerTeleop\":\"" + speakerTeleop + "\"," + "\"noteAcquired\":\"" + noteAcquired + "\"," + "\"chaining\":\"" + chaining + "\","
+                + "\"trapSucess\":\"" + trapSucess + "\"," + "\"trapFail\":\"" + trapFail + "\"," + "\"harmo\":\"" + harmo + "\"," + "\"humanPlayerAtAmp\":\"" + humanPlayerAtAmp + "\","
+                + "\"humanPlayerNotesThrown\":\"" + humanPlayerNotesThrown + "\"," + "\"humanPlayerNotesThrown\":\"" + humanPlayerNotesThrown + "\"," + "\"notes\":\"" + notes + "\"}";
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/MatchData.java:58]: If there is any `"` character in the notes, the file will have invalid JSON and won't be able to be read. The best solution is indeed to use a JSON library, but if you aren't going to do that, we can at least sanitize the notes, by replacing `"` with `\"` or `\u0020`.

**Review by carlosfigueira — APPROVED** [review]
:shipit:

**Pull request merged**